### PR TITLE
WebGL encapsulation

### PR DESCRIPTION
**Pull request by kvark**
Refactors our WebGL usage and fixes a regression from #1509
also related to #1353 (should make it easier)

r? @MortimerGoro 
cc @glennw

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1530)

### CODE REVIEW

**Comment by MortimerGoro** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/webrender/1530#-:-Kq8pAo_srV0EgOHCBzK:bnfp4nl)*

**Comment by kvark** [comment]
not sure if this is going to work
@bors-servo r=MortimerGoro

File: webrender/src/render_backend.rs [hunk 1]
```diff
@@ -105,6 +105,59 @@ impl Document {
     }
 }
 
+struct WebGL {
+    last_id: WebGLContextId,
+    contexts: HashMap<WebGLContextId, GLContextWrapper, BuildHasherDefault<FnvHasher>>,
```
Comment (emilio) [inline webrender/src/render_backend.rs]: nit: Could use `fnv::FnvHashMap`, and `HashMap::default()` below.

Comment (kvark) [inline webrender/src/render_backend.rs]: I just followed the existing code. Would be great to convert it in bulk to a shorter form as you propose, just in a separate PR

Comment (MortimerGoro) [inline webrender/src/render_backend.rs]: Th WebGL code in WR will be deleted after https://github.com/servo/servo/pull/17891 lands, so it's not worth the effort to work on that.

Comment (kvark) [inline webrender/src/render_backend.rs]: @MortimerGoro wow, +3,138 −1,417

**Pull request merged**